surefire-booter/src/main/java/org/apache/maven/surefire/booter/spi/SurefireMasterProcessChannelProcessorFactory.java

```diff
@@ -119,4 +128,23 @@ private final void setTrueOptions( SocketOption<Boolean>... options )
             }
         }
     }
+
+    private static String parseSessionId( URI uri )
```

Comment:
I think `extractSessionId` is a better name since the URI has already been parsed, we just need to extract the session id. Thoogh, this might be just a personal taste.

Reply:
done